flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionFactory.java

```diff
@@ -187,7 +197,8 @@ private static void releasePartitionsQuietly(ResultSubpartition[] partitions, in
 				numberOfSubpartitions * networkBuffersPerChannel + floatingNetworkBuffersPerGate : Integer.MAX_VALUE;
 			// If the partition type is back pressure-free, we register with the buffer pool for
 			// callbacks to release memory.
-			return bufferPoolFactory.createBufferPool(numberOfSubpartitions,
+			return bufferPoolFactory.createBufferPool(
+				numberOfSubpartitions + 1,
```

Comment:
I think this needs to be squashed with the previous commit as again, those two things are heavily interconnected and on their own they are incomplete/missing context.

Also, can we add a test for that? That `StreamTask` is able to make a non-blocking progress with a minimal `LocalBufferPool`? Or is it already covered by some existing tests?

Reply:
You are right, and my previous thought was also for squashing these two commits together finally, because it would cause potential stuck problem if only one commit merged. It is only for review purpose to make it separate now.
I planed to add new tests for cover the changes, and it seems no previous tests to verify the minimum size of pool.